DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/DelegateGroups/DelegateGroupsViewModelFilterOptions.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.DelegateGroups
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Models.CustomPrompts;
+    using DigitalLearningSolutions.Data.Models.DelegateGroups;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Helpers.FilterOptions;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+
+    public static class DelegateGroupsViewModelFilterOptions
+    {
+        public static readonly IEnumerable<FilterOptionViewModel> BaseLinkedFieldOptions = new[]
+        {
+            DelegateGroupLinkedFieldFilterOptions.None,
+            DelegateGroupLinkedFieldFilterOptions.JobGroup
+        };
+
+        public static IEnumerable<FilterOptionViewModel> GetLinkedFieldOptions(
+            IEnumerable<CustomPrompt> registrationPrompts
+        )
+        {
+            var promptOptions = registrationPrompts.Select(
+                prompt => new FilterOptionViewModel(
+                    prompt.CustomPromptText,
+                    nameof(Group.LinkedToField) + FilteringHelper.Separator + nameof(Group.LinkedToField) +
+                    FilteringHelper.Separator + GetLinkedFieldIdFromRegistrationPromptNumber(prompt.CustomPromptNumber),
+                    FilterStatus.Default
+                )
+            );
+
+            return BaseLinkedFieldOptions.Concat(promptOptions);
+        }
+
+        public static IEnumerable<FilterOptionViewModel> GetAddedByOptions(
+            IEnumerable<(int adminId, string adminName)> admins
+        )
+        {
+            return admins.Select(
+                admin => new FilterOptionViewModel(
+                    admin.adminName,
+                    nameof(Group.AddedByAdminId) + FilteringHelper.Separator + nameof(Group.AddedByAdminId) +
+                    FilteringHelper.Separator + admin.adminId,
+                    FilterStatus.Default
+                )
+            );
+        }
+
+        // Centre registration prompts correspond to Groups.LinkedToField as follows:
+        // 1, 2 and 3 correspond to 1, 2 and 3 respectively
+        // 4, 5 and 6 correspond to 5, 6 and 7 respectively

```

Comment:
This comment seems a bit unnecessary as it just describes what the function is doing, but not why. It might be better to say that LinkedToField = 4 is reserved for JobGroup, so has to be skipped when mapping prompts.